Modules/Filtering/MathematicalMorphology/test/itkTopHatImageFilterTest.cxx

```diff
@@ -24,14 +24,68 @@
 #include "itkImageFileWriter.h"
 #include "itkTestingMacros.h"
 
+
+template <typename TKernelImageFilter>
+int
+itkTopHatImageFilterTestHelper(TKernelImageFilter *                    filter,
+                               typename TKernelImageFilter::KernelType ball,
+                               const bool                              safeBorder,
+                               const int                               algorithm,
+                               const bool                              forceAlgorithm,
+                               const char *                            inputFileName,
+                               const char *                            outputFileName)
+{
+  // Declare the reader and writer
+  using ReaderType = itk::ImageFileReader<TKernelImageFilter::InputImageType>;
+  ReaderType::Pointer reader = ReaderType::New();
+  reader->SetFileName(inputFileName);
+
+  itk::SimpleFilterWatcher watcher(filter, "filter");
+
+  ITK_TEST_SET_GET_BOOLEAN(filter, SafeBorder, safeBorder);
+
+  filter->SetAlgorithm(algorithm);
+  ITK_TEST_SET_GET_VALUE(algorithm, filter->GetAlgorithm());
+
+  ITK_TEST_SET_GET_BOOLEAN(filter, ForceAlgorithm, forceAlgorithm);
+
+  // Connect the structuring element
+  filter->SetKernel(ball);
+
+  // Connect the pipeline
+  filter->SetInput(reader->GetOutput());
+
+  ITK_TRY_EXPECT_NO_EXCEPTION(filter->Update());
+
+
+  // Write output
+  using WriterType = itk::ImageFileWriter<TKernelImageFilter::OutputImageType>;
+  WriterType::Pointer writer = WriterType::New();
```

Comment:
Also add `typename` here.

Reply:
Fixed in 18d1de6. Thanks for spotting these !